### ISSUE TITLE
Advanced Javascript Events

### ISSUE THREAD

**Issue by AaronShea** [AaronShea/BLUI#46]
I'll be used this issue as a scratchpad for improving BLUI's Javascript integration.

- [ ] Support named/custom events to avoid "name drilldown" in code
- [ ] Support more types for return values
- [ ] Nested JSON improvements
- [ ] Live bindings?

[Updated 8/3/2015]

**Comment by The-Skas** [comment]
Hey there,

Awesome progress as always.

In the browser, the blu_event(msg,value) method is great for sending a message across to unreal, but I was wondering if its possible to send a message and have UE4 return a value? If not, maybe its something I could look into. 

--------
Browser JS
-------
//JS should wait until blu_event returns a value.
var x = blu_event("getActor", "Door_c_1");
console.log(x);

**Comment by AaronShea** [comment]
Sorry for the late reply, Github didn't notify me of this issue comment for some reason!

That's an interesting idea, I'll have to toy with how that could be implemented.
As of now they are using inter-process messages between the BluBrowser and UE4. When blu_event is called it simply fires a message off to the UE4 process and returns true. 

I've been planning the JS update for a while now, so we'll see where it goes!

**Comment by The-Skas** [comment]
Thank you for replying :-)

I was looking for the blu_event method in the release code to see if I could modify it, yet I couldnt seem to find it.

 Could you point to the relevant line of code if you may?

**Comment by AaronShea** [comment]
Indeed, I'm in the process of working on a build script that will build all of BLUI for you, no messing about with grabbing CEF yourself and compiling ect

**Comment by ImmanuelSamuel** [comment]
@AaronShea Are you still planning to implement this feature in the near future?

**Comment by AaronShea** [comment]
I have not been able to dedicate time to BLUI recently due to other commitments in life. I'm still around though, it's certainly a nice feature, but with college starting back up again soon I'm not sure when I'll get to it
